modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -654,10 +654,16 @@ def to_pandas(cls, partitions):
             axis = 1
         else:
             ErrorMessage.catch_bugs_and_request_email(True)
+
+        def is_part_empty(part):
+            return part.empty and (
+                not isinstance(part, pandas.DataFrame) or (len(part.columns) == 0)
+            )
+
         df_rows = [
             pandas.concat([part for part in row], axis=axis)
             for row in retrieved_objects
-            if not all(part.empty for part in row)
+            if not all(is_part_empty(part) for part in row)
```

Comment:
Have we had the issue for `PandasOn<Engine>` execution too? If so, we should also add a test for `PandasOn<Engine>`.